nemoguardrails/cli/__init__.py

```
@@ -243,6 +250,18 @@ def action_server(
 ):
     """Start a NeMo Guardrails actions server."""
 
+    try:
+        import fastapi  # noqa: F401
+        import uvicorn
+    except ImportError:
+        typer.secho(
+            "Server dependencies are missing. Install them with: pip install nemoguardrails[server]",
+            fg=typer.colors.RED,
+        )
+        raise typer.Exit(1)
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)